src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -39,8 +45,26 @@ public void testOnlySetContext(SessionContext sessionContext) {
     this.sessionContext = sessionContext;
   }
 
+
+  //method to override AbuseDetectionFeature with Mock FOR TESTING        
+  public void testOnlySetAbuseDetection(AbuseDetection abuseFeature) {
+    this.abuseDetectionFeature = abuseFeature;
+  }
+
+
+  /*
+  * This method relates to rate limiting problem by analyzing the time of the 
+  * request and limiting the amount of requests coming in. This is a problem
+  * because malicious users can overload requests to the backend, so we created
+  * this method to help limit the amount of requests coming to the backend.
+  */
+  public boolean analyzeTimeOfRequest(HttpServletRequest request) {
+    return abuseDetectionFeature.addRequest(new Date());
```

Comment:
What will happen if the abuseDetectionFeature is not initialized?

Reply:
Then it wouldn't be able to add the request

Reply:
Do you have a test for this?

Reply:
No. I'll make one.

Reply:
Added test and conditionals.

Reply:
Thanks for adding the tests. So what is the plan for enabling this feature on the website? It seems like the abuseDetectionFeature is only ever initialized in tests.

Reply:
What do you mean by enabling to the website, isn't this a backend only application?

Reply:
I guess the plan to enable this feature is have this feature used only when users send a request to get match requests from the logged_in_homepage.